server/index-node/src/resolver.rs

```diff
@@ -470,17 +469,133 @@ where
                 data.get_required::<q::Value>("subgraphDeploymentAssignments")
                     .expect("missing deployment assignments"),
-            ),
-        ]);
+        };
 
         Ok(IndexingStatuses::from(transformed_data).into())
     }
+
+    fn resolve_proof_of_indexing(
+        &self,
+        argument_values: &HashMap<&q::Name, q::Value>,
+    ) -> Result<q::Value, QueryExecutionError> {
+        let request = argument_values
+            .get_required::<ProofOfIndexingInput>("request")
+            .expect("No request provided");
+
+        let ProofOfIndexingInput {
+            ethereum_block_number,
+            subgraph_name,
+        } = request;
+
+        // Build a query matching the current deployment (if any) for the given id.
+        let query = Query {
+            // The query is against the subgraph of subgraphs
+            schema: self
+                .store
+                .api_schema(&SUBGRAPHS_ID)
+                .map_err(QueryExecutionError::StoreError)?,
+
+            // We're querying the deployment id of the current version for the given subgraph id.
+            document: q::parse_query(
+                r#"
+                query subgraphs($name: String!) {
+                    subgraphs(where: {name: $name}, first: 1) {
+                        currentVersion {
+                            deployment {
+                                id
+                            }
+                        }
+                    }
+                }
+                "#,
+            )
+            .unwrap(),
+
+            variables: Some(QueryVariables::new(HashMap::from_iter(
+                vec![("name".into(), q::Value::String(subgraph_name.clone()))].into_iter(),
+            ))),
+        };
+
+        // Execute the query
+        let result = self
+            .graphql_runner
+            .run_query_with_complexity(query, None, None, Some(std::u32::MAX))
+            .wait()
+            .expect("error querying subgraph deployments");
+
+        let data = match result.data {
+            Some(data) => data,
+            None => {
+                error!(
+                    self.logger,
+                    "Failed to query subgraph deployments";
+                    "subgraph" => subgraph_name,
+                    "errors" => format!("{:?}", result.errors)
+                );
+                return Ok(q::Value::Null);
+            }
+        };
+
+        let subgraphs = match data
+            .get_optional::<q::Value>("subgraphs")
+            .expect("invalid subgraphs")
+        {
+            Some(subgraphs) => subgraphs,
+            None => return Ok(q::Value::Null),
+        };
+
+        let mut subgraphs = subgraphs
+            .get_values::<q::Value>()
+            .expect("invalid subgraph values");
+
+        let subgraph = if let Some(subgraph) = subgraphs.pop() {
+            subgraph
+        } else {
+            return Ok(q::Value::Null);
+        };
```

Comment:
Most of this we can get rid of when we query by subgraph ID.

Reply:
I changed it to subgraph id, but unless I'm misunderstanding all this code stays. We still need to go from a subgraph id to a deployment id.

Reply:
Subgraph ID = deployment ID 😉